src/enkf/gridio_fv3reg.f90

```diff
@@ -659,496 +661,1825 @@ subroutine writegriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,vargrid
+              do k=1,nlevs
+               nn = nn_tile0
+               do j=1,ny_res
+               do i=1,nx_res
+                   nn=nn+1
+                   workinc3d(i,j,k)=vargrid(nn,levels(w_ind-1)+k,nb,ne)
+               enddo
+               enddo
+              enddo
+              workvar3d=workvar3d(1:nx_res,:,:)+workinc3d
+           endif
+           do k=1,nlevs
+              call mpi_scatterv(workvar3d(:,:,k),recvcounts2d,displs2d,mpi_real4,&
+              locworkvar3d(:,:,k),recvcounts2d(iope+1),mpi_real4,0,iocomms(mem_pe(nproc)),ierror)
+           enddo
+           call write_fv3_restart_data3d(varstrname,fv3filename,file_id,locworkvar3d)
 
-    endif
-    if (oz_ind > 0) then
-       varstrname = 'o3mr'
-         
-       call fv3lamfile%get_idfn(varstrname,file_id,fv3filename)
-       call read_fv3_restart_data3d(varstrname,fv3filename,file_id,workvar3d)
-      do k=1,nlevs
-          nn = nn_tile0
-      do j=1,ny_res
-         do i=1,nx_res
-            nn=nn+1
-            workinc3d(i,j,k)=vargrid(nn,levels(oz_ind-1)+k,nb,ne) 
-         enddo
-      enddo
-      enddo
-      workvar3d=workvar3d+workinc3d
-       call write_fv3_restart_data3d(varstrname,fv3filename,file_id,workvar3d)
+        endif
 
-    endif
+        if (tv_ind > 0.or.tsen_ind>0 ) then
+             
+           allocate(locqworkvar3d(nxloc,nyloc,nlevs))
+           varstrname = 'T'
+           if(tsen_ind>0) then
+             call fv3lamfile%get_idfn(varstrname,file_id,fv3filename)
+             call read_fv3_restart_data3d(varstrname,fv3filename,file_id,locworkvar3d)
+             do k=1,nlevs
+               call mpi_gatherv(locworkvar3d(1:nxloc,1:nyloc,k), recvcounts2d(iope+1), mpi_real4, &
+                       workvar3d(:,:,k), recvcounts2d, displs2d,&
+                       mpi_real4, 0,iocomms(mem_pe(nproc)) ,iret)
+             enddo
+             if(iope ==0 ) then
+               do k=1,nlevs
+                  nn = nn_tile0
+                  do j=1,ny_res
+                  do i=1,nx_res
+                     nn=nn+1
+                     workinc3d(i,j,k)=vargrid(nn,levels(tsen_ind-1)+k,nb,ne) 
+                  enddo
+                  enddo
+               enddo
+                   workvar3d=workvar3d+workinc3d
+             endif
+             do k=1,nlevs
+                call mpi_scatterv(workvar3d(:,:,k),recvcounts2d,displs2d,mpi_real4,&
+                locworkvar3d(:,:,k),recvcounts2d(iope+1),mpi_real4,0,iocomms(mem_pe(nproc)),ierror)
+             enddo
+             call write_fv3_restart_data3d(varstrname,fv3filename,file_id,locworkvar3d)
+           else  ! tv_ind >0  
+             if(iope ==0) then
+               do k=1,nlevs
+                 nn = nn_tile0
+                 do j=1,ny_res
+                 do i=1,nx_res
+                    nn=nn+1
+                    workinc3d(i,j,k)=vargrid(nn,levels(tv_ind-1)+k,nb,ne) 
+                 enddo
+                 enddo
+               enddo
+              endif
```

Comment:
Please align line 2082 with line 2072 and align all following lines appropriately.